src/axiom_py/query/result.py

```diff
@@ -151,17 +141,122 @@ class QueryLegacyResult:
     savedQueryID: Optional[str] = field(default=None)
 
 
+@dataclass
+class Source:
+    name: str
+
+
+@dataclass
+class Order:
+    desc: bool
+    field: str
+
+
+@dataclass
+class Group:
+    name: str
+
+
+@dataclass
+class Range:
+    # Start is the starting time the query is limited by.
+    start: datetime
+    # End is the ending time the query is limited by.
+    end: datetime
+    # Field specifies the field name on which the query range was restricted.
+    field: str
+
+
+@dataclass
+class Aggregation:
+    # Args specifies any non-field arguments for the aggregation.
+    args: Optional[List[object]]
+    # Fields specifies the names of the fields this aggregation is computed on.
+    fields: Optional[List[str]]
+    # Name is the system name of the aggregation.
+    name: str
+
+
+@dataclass
+class Field:
+    name: str
+    type: str
+    agg: Optional[Aggregation]
+
+
+@dataclass
+class Bucket:
+    # Field specifies the field used to create buckets on.
+    field: str
+    # An integer or float representing the fixed bucket size.
+    size: Union[int, float]
+
+
+@dataclass
+class Table:
+    # Name is the name assigned to this table. Defaults to "0".
+    # The name "_totals" is reserved for system use.
+    name: str
+    # Sources contain the names of the datasets that contributed data to these
+    # results.
+    sources: List[Source]
+    # Order echoes the ordering clauses that was used to sort the results.
+    order: List[Order]
+    # Groups specifies which grouping operations has been performed on the
+    # results.
+    groups: List[Group]
+    range: Range
```

Comment:
range is optional
```suggestion
    range: Optional[Range]
```

Reply:
Fixed in [cbb7584](https://github.com/axiomhq/axiom-py/pull/136/commits/cbb7584df0c782749daad2f68f5f28a71fe865f8) among other required/optional changes